src/pages/api/multipleProductValidity.ts

```diff
@@ -0,0 +1,130 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import Cookies from 'cookies';
+import {
+    NUMBER_OF_PRODUCTS_COOKIE,
+    MULTIPLE_PRODUCT_COOKIE,
+    OPERATOR_COOKIE,
+    CSV_ZONE_UPLOAD_COOKIE,
+    PERIOD_SINGLE_OPERATOR_SERVICES_COOKIE,
+    PERIOD_TYPE_COOKIE,
+    MATCHING_DATA_BUCKET_NAME,
+} from '../../constants/index';
+import { isSessionValid } from './service/validator';
+import { redirectToError, setCookieOnResponseObject, getDomain, redirectTo } from './apiUtils';
+import { Product } from '../multipleProductValidity';
+import { getCsvZoneUploadData, putStringInS3 } from '../../data/s3';
+import { batchGetStopsByAtcoCode, Stop } from '../../data/auroradb';
+
+interface DecisionData {
+    operatorName: string;
+    type: string;
+    nocCode: string;
+    products: Product[];
+    selectedServices?: [];
+    zoneName?: string;
+    stops?: Stop[];
+}
+
+export const addErrorsIfInvalid = (req: NextApiRequest, numberOfProducts: number, products: Product[]): Product[] => {
+    const limiter = new Array(numberOfProducts);
+    const response: Product[] = [];
+    for (let i = 0; i < limiter.length; i += 1) {
```

Comment:
Rather than use this limiter and for loop, could we map over the `products` array instead?

Reply:
Did not realise until now that you can access an iterator using map 🤦